packages/elements/src/components/link-with-icon/link-with-icon.ts

```diff
@@ -0,0 +1,45 @@
+/**
+ * @license
+ *
+ * Copyright IBM Corp. 2020
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { html, customElement } from 'lit-element';
+import settings from 'carbon-components/es/globals/js/settings';
+import styles from '@carbon/ibmdotcom-styles/scss/components/link-with-icon/_link-with-icon.scss';
+import ddsSettings from '@carbon/ibmdotcom-utilities/es/utilities/settings/settings';
+import DDSLink from '../../globals/internal/link';
+
+const { prefix } = settings;
+const { stablePrefix: ddsPrefix } = ddsSettings;
+
+/**
+ * Link with icon.
+ *
+ * @element dds-link-with-icon
+ */
+@customElement(`${ddsPrefix}-link-with-icon`)
```

Comment:
More of a general question, and might even be a good huddle topic. Can this be done with ES6 too instead of typescript, and are there benefits to going with typescript over ES6? I want to choose something that the team would be comfortable with as well.

Reply:
@jeffchew Great question, we can definitely write components in non-TS. I started with TS from two reasons:

* The underlying library, [`lit-element`](https://www.npmjs.com/package/lit-element), is written in TS
* Unlike Carbon core, `ibm-dotcom-library` has components that deals with complex data. TS allows us to define data scheme (like below) and detect errors with missing/wrong data:

```typescript
/**
 * Base CTA data item.
 */
export interface BASE_CTA_ITEM {
  /**
   * The copy text.
   */
  copy: string;

  /**
   * The default file name for downloading, should be used only with `CTA_TYPE.DOWNLOAD`.
   */
  download: string;

  /**
   * The href. For video CTA type, the video ID should be given.
   */
  href: string;

  /**
   * The CTA type.
   */
  type: CTA_TYPE;
}
```

But I'm open on this topic to see which choice our team would like - So let's keep the thoughts coming!